narwhal/primary/src/block_synchronizer/tests/block_synchronizer_tests.rs

```diff
@@ -727,11 +737,13 @@ async fn test_reply_with_payload_already_in_storage_for_own_certificates() {
     for _ in 0..5 {
         let batch = fixture_batch_with_transactions(10);
 
-        let header = primary
-            .header_builder(&committee)
-            .with_payload_batch(batch.clone(), 0, 0)
-            .build()
-            .unwrap();
+        let header = Header::V1(
+            primary
+                .header_builder(&committee)
+                .with_payload_batch(batch.clone(), 0, 0)
+                .build()
```

Comment:
it might make more sense to have builder do the wrapping and return the appropriate version. You will have to somehow get the ProtocolConfig to do that, of course. If you have access to the sui committee, you can get the right config.

This may be a lot of work - it can be defered until we actually have to do an upgrade.